src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala

```diff
@@ -1,24 +1,34 @@
 package com.linkedin.feathr.offline.config.location
 
+import com.fasterxml.jackson.annotation.JsonAlias
 import com.fasterxml.jackson.module.caseclass.annotation.CaseClassDeserialize
 import com.linkedin.feathr.offline.source.dataloader.jdbc.JdbcUtils
 import com.linkedin.feathr.offline.source.dataloader.jdbc.JdbcUtils.DBTABLE_CONF
 import org.apache.spark.sql.{DataFrame, SparkSession}
 import org.eclipse.jetty.util.StringUtil
 
 @CaseClassDeserialize()
-case class Jdbc(url: String, dbtable: String, user: String = "", password: String = "", token: String = "", useToken: Boolean = false, anonymous: Boolean = false) extends InputLocation {
+case class Jdbc(url: String, @JsonAlias(Array("query")) dbtable: String, user: String = "", password: String = "", token: String = "", useToken: Boolean = false, anonymous: Boolean = false) extends InputLocation {
   override def loadDf(ss: SparkSession, dataIOParameters: Map[String, String] = Map()): DataFrame = {
     var reader = ss.read.format("jdbc")
       .option("url", url)
     if (StringUtil.isBlank(dbtable)) {
       // Fallback to default table name
       reader = reader.option("dbtable", ss.conf.get(DBTABLE_CONF))
     } else {
-      reader = reader.option("dbtable", dbtable)
+      val q = dbtable.trim
+      if("\\s".r.findFirstIn(q).nonEmpty) {
+        // This is a SQL instead of a table name
+        reader = reader.option("query", q)
+      } else {
+        reader = reader.option("dbtable", q)
+      }
     }
     if (useToken) {
-      reader.option("accessToken", LocationUtils.envSubstitute(token)).load
+      reader.option("accessToken", LocationUtils.envSubstitute(token))
+        .option("hostNameInCertificate", "*.database.windows.net")
+        .option("encrypt", true)
```

Comment:
document why it's always true

Reply:
These properties are required by AzureSQL Spark connector, they're set according to Azure docs at https://docs.microsoft.com/en-us/azure/azure-sql/database/spark-connector#connecting-using-an-access-token